modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/CatalogIndexDescriptor.java

```diff
@@ -34,20 +34,25 @@ public abstract class CatalogIndexDescriptor extends CatalogObjectDescriptor {
     /** Index descriptor type. */
     private final CatalogIndexDescriptorType indexType;
 
+    /** Flag indicating that this index has been created at the same time as its table. */
+    private final boolean createdWithTable;
```

Comment:
frankly, I don't get why do we need this attribute. Let's add a comment explaining the reason for this flag, as well as an example of cases of who can/should observe it and why.